01_fib/luliangLi/main_test.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+)
+
+func TestFibOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	testCases := map[string]struct {
+		input int
+		want  string
+	}{
+		"7":  {input: 7, want: "1\n1\n2\n3\n5\n8\n13\n"},
+		"-7": {input: -7, want: "1\n-1\n2\n-3\n5\n-8\n13\n"},
+		"0":  {input: 0, want: "0\n"},
+		"1":  {input: 1, want: "1\n"},
+		"-1": {input: -1, want: "1\n"},
+		"2":  {input: 2, want: "1\n1\n"},
+		"-2": {input: -2, want: "1\n-1\n"},
+	}
+
+	for name, test := range testCases {
+		// t.Run creates a sub test and runs it like a normal test
+		test := test
```

Comment:
Not needed as a variable assigns to itself.

Reply:
Thanks for reviewing my code. 
Not sure if you encountered the issue or not if removing the tricky "test := test", while I got the following issue anyhow:
Using the variable on range scope `test` in function literal (scopelint)
			fib(test.input)

So I googled it and found the solution in the link: https://github.com/kyoh86/scopelint., which taught me to add the line to pin an external variable to an internal one.

Reply:
Sorry LuliangLi, my bad here.